src/Yaml.zig

```diff
@@ -175,12 +175,20 @@ fn parseStruct(self: Yaml, arena: Allocator, comptime T: type, map: Map) Error!T
     var parsed: T = undefined;
 
     inline for (struct_info.fields) |field| {
-        const value: ?Value = map.get(field.name) orelse blk: {
+        var value: ?Value = map.get(field.name) orelse blk: {
             const field_name = try mem.replaceOwned(u8, arena, field.name, "_", "-");
             break :blk map.get(field_name);
         };
 
         if (@typeInfo(field.type) == .optional) {
+            const maybe_default_value: ?field.type = field.defaultValue();
+            if (value == null and maybe_default_value != null) {
+                if (Value.encode(arena, maybe_default_value.?)) |encoded_default_value| {
+                    value = encoded_default_value;
+                } else |err| {
+                    log.debug("default value encoding error {} for: {s} falling back to null", .{ err, field.name });
+                }
+            }
```

Comment:
I think this would be more idiomatic

```suggestion
            if (value == null) blk: {
                const maybe_default_value = field.defaultValue() orelse break :blk;
                value = Value.encode(arena, maybe_default_value) orelse break :blk;
            }
```

I also don't see any value in having `log.debug` statement here.